include/seastar/http/internal/content_source.hh

```diff
@@ -40,10 +40,15 @@ namespace internal {
  * */
 class content_length_source_impl : public data_source_impl {
     input_stream<char>& _inp;
+    http::reply* _rep;
     size_t _remaining_bytes = 0;
 public:
+    content_length_source_impl(input_stream<char>& inp, http::reply* rep)
+        : _inp(inp), _rep(rep), _remaining_bytes(_rep->content_length) {
+    }
+
     content_length_source_impl(input_stream<char>& inp, size_t length)
-        : _inp(inp), _remaining_bytes(length) {
+        : _inp(inp), _rep(nullptr), _remaining_bytes(length) {
```

Comment:
Is this constructor still in use?

Reply:
yes, in `make_content_stream`, it has its own way to take care of unconsumed bytes in `connection::read_one`